torch/distributed/_tensor/_dispatch.py

```diff
@@ -71,6 +71,13 @@ class OpDispatcher:
     Op dispatching class instance to handle args/kwargs pre-processing (un-wrapping), sharding
     propagation, redistribute local args, local compute, and post-processing (re-wrapping). It
     also handles any op specific logic if necessary.
+
+    NOTE: Given the runtime overhead of Tensor subclass (__torch_dispatch__), the OpDispatcher
+    is designed to minimize the CPU overhead by using the tricks of proper unflattening, faster
+    pytree if needed, and leveraging various caching mechanisms implemented in the sharding
+    propagation and redistribute modules. The CPU overhead is critical to eager mode performance,
+    one need to carefully measure the CPU overhead when making significant changes to the
+    OpDispatcher and ShardingPropagator.
```

Comment:
is there any suggestion on CPU overhead measurement? like, by running profiling in model testing?

Reply:
yeah running profiling with some benchmark scripts to measure runtime overhead